src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,44 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+  constructor(health = 100, name) {
+    this.health = health;
+    this.name = name;
+    Animal.alive.push(this);
+  }
+
+  checkHealth() {
+    if (this.health <= 0) {
+      this.index = Animal.alive.indexOf(this);
+      Animal.alive.splice(this.index, 1);
+    }
+  }
```

Comment:
[CODE PERFOMANCE] - The current implementation of removing dead animals from the `Animal.alive` array is using nested loops (`indexOf` + `splice`). Consider using `filter` instead to improve performance and code clarity.